### PR TITLE
DAOS-11909 test: trigger rebuild

### PR DESCRIPTION
**Pull request by daltonbohning**
- Update packaging
- Trigger rebuild

Signed-off-by: Dalton Bohning <dalton.bohning@intel.com>

### CODE REVIEW

**Comment by daltonbohning** [comment]
All mpifileutils tests passed on master: https://build.hpdd.intel.com/blue/organizations/jenkins/daos-stack%2Fdaos/detail/PR-10571/1/pipeline/
All mpifileutils tests passed in release/2.2: https://build.hpdd.intel.com/blue/organizations/jenkins/daos-stack%2Fdaos/detail/PR-10572/1/pipeline/

**Comment by daltonbohning** [comment]
@brianjmurrell Does the RPM lint need to be resolved before landing this? All this does is trigger a rebuild to resolve https://daosio.atlassian.net/browse/DAOS-11909

File: Jenkinsfile
Comment (daltonbohning) [inline Jenkinsfile:43]: Is this okay?

**Review by brianjmurrell — CHANGES_REQUESTED** [review]
A rebuild without bumping the `Release:` and/or `Version:` should never be done.  We in fact should add a check to the pipeline for dependencies to fail a PR that does not bump one of them.

The bigger issue however is the ongoing discussion of the need to rebuild this because the builds in our repos no longer work with the existing DAOS client and whether the SOVERSION of the DAOS client was bumped to reflect that.

So -1 for now until the above two issue are resolved.

**Comment by mchaarawi** [comment]
> The bigger issue however is the ongoing discussion of the need to rebuild this because the builds in our repos no longer work with the existing DAOS client and whether the SOVERSION of the DAOS client was bumped to reflect that.

this is true, however the client change that was made is only being used here in mfu. i think we should just land this update to prevent weekly tests from failing. If you really insist on the major verison bump, we can discuss it in a different meeting like the PDT, but I am aware of this change and don't think it is required because no one really is using MFU in production at this point and this is where it's only being used.
so would appreciate if you unblock this for now.

File: Jenkinsfile
Comment (jolivier23) [inline Jenkinsfile:43]: Can we remove centos7 here?   In 2.2, we still build and test on centos7

Comment (jolivier23) [reply Jenkinsfile:43]: I guess old packaging will still be there so maybe it's fine.

Comment (daltonbohning) [reply Jenkinsfile:43]: That's what I thought, but someone correct me if this is wrong :)

**Comment by brianjmurrell** [comment]
> i think we should just land this update to prevent weekly tests from failing.

We cannot.  If we rebuild mpifileutils without bumping the SOVERSION on the DAOS client, this new mpifileutils build will be used for DAOS builds prior to the ABI breakage and we'd have the inverse problem that we have now (the new mpifileutils will break when used with pre-ABI-breaking DAOS builds).  Moreover, this will include even the *current* DAOS builds on any branches that did not have the same ABI breaking update such as 2.2 and 2.0.

We'd effectively be breaking 2.0 and 2.2 by hackishly fixing 2.4/master.

**Comment by mchaarawi** [comment]
> > i think we should just land this update to prevent weekly tests from failing.
> 
> We cannot. If we rebuild mpifileutils without bumping the SOVERSION on the DAOS client, this new mpifileutils build will be used for DAOS builds prior to the ABI breakage and we'd have the inverse problem that we have now (the new mpifileutils will break when used with pre-ABI-breaking DAOS builds). Moreover, this will include even the _current_ DAOS builds on any branches that did not have the same ABI breaking update such as 2.2 and 2.0.
> 
> We'd effectively be breaking 2.0 and 2.2 by hackishly fixing 2.4/master.

no we would not be breaking DAOS 2.2 and 2.0 because for this change the size of the structure has increased. so MFU giving a bigger struct to DAOS 2.2 and 2.0 is OK and will not break anything.
again i've considered these issues already and i don't think the bump is necessary in this case.

**Comment by brianjmurrell** [comment]
If the rpmlinting issues that are breaking the build can be resolved, we can get some testing on all three branches to confirm.

**Comment by mchaarawi** [comment]
> If the rpmlinting issues that are breaking the build can be resolved, we can get some testing on all three branches to confirm.

@daltonbohning already has a PR above where he tested against 2.2
2.0 would be in a similar situation as 2.2 and shouldn't be hard to test, but 2.0 is EOL on our side so not sure why you are still concerned with it..

File: Jenkinsfile
Comment (brianjmurrell) [inline Jenkinsfile:43]: Please use the [`packageBuildingPipelineDAOSTest()`](https://github.com/daos-stack/pipeline-lib/blob/master/vars/packageBuildingPipelineDAOSTest.groovy) function here so we have some per-branch regression/compatibility testing for this package.

**Review by brianjmurrell — CHANGES_REQUESTED** [review]

**Comment by brianjmurrell** [comment]
> @daltonbohning already has a PR above where he tested against 2.2:

Since I have already requested using the `packageBuildingPipelineDAOSTest()` function for this package, can I reserve my +1 until that is done and passing since it has to pass to land in any case?

**Comment by daltonbohning** [comment]
> Since I have already requested using the `packageBuildingPipelineDAOSTest()` function for this package, can I reserve my +1 until that is done and passing since it has to pass to land in any case?

I think that's reasonable - thanks!

**Comment by mchaarawi** [comment]
> > @daltonbohning already has a PR above where he tested against 2.2:
> 
> Since I have already requested using the `packageBuildingPipelineDAOSTest()` function for this package, can I reserve my +1 until that is done and passing since it has to pass to land in any case?

Indeed, we have added something similar for the hdf5 and hdf5-vol package. This helps in testing without needing to push separate prs to just test the update.

File: Jenkinsfile
Comment (mchaarawi) [inline Jenkinsfile:43]: The idea for this is to use a test tag for mpifileutils tests to be tested. Something like:
https://github.com/daos-stack/hdf5-vol-daos/blob/master/Jenkinsfile#L43
but different tag of course

Comment (brianjmurrell) [reply Jenkinsfile:43]: Ah, yes indeed!  Great catch @mchaarawi!

@daltonbohning You'd want to use a tag that (at least) captures the (i.e. weekly) test that is failing.  You can use more than one tag IIRC, just like `Test-tag:`

**Comment by brianjmurrell** [comment]
@daltonbohning You are going to need to resolve the rpmlint errors (at least) in order for the pipeline to run the testing stage.  Looks like the errors are:

```
mpifileutils-mpich-devel.x86_64: E: static-library-without-debuginfo /usr/lib64/mpich/lib/libmfu.a
mpifileutils-openmpi-devel.x86_64: E: static-library-without-debuginfo /usr/lib64/openmpi/lib/libmfu.a
```

Those static libraries are typically simply not included in RPMs.  You can remove them in `%install` and then remove any references that may be in the `*-devel` `%files` sections.

**Comment by daltonbohning** [comment]
Thanks @brianjmurrell @mchaarawi. I ported the lint fixes from #27, and included the `mfu` tag which will run ALL tests using mpifileutils.

File: mpifileutils.spec
Comment (brianjmurrell) [inline mpifileutils.spec:86]: Please set this to 1 as it seems we have changed our policy/desire for wanting debuginfo subpackages recently.

```suggestion
%global _debuginfo_subpackages 1
```

**Review by brianjmurrell — CHANGES_REQUESTED** [review]

**Comment by daltonbohning** [comment]
A single 2.2 test failed: https://build.hpdd.intel.com/blue/organizations/jenkins/daos-stack%2Fdaos/detail/ci-daos-stack-mpifileutils-pkg-PR-30-release-2.2/4/tests
For a known reason: https://daosio.atlassian.net/browse/DAOS-10258

**Comment by daltonbohning** [comment]
Side note: @brianjmurrell Should packageBuildingPipelineDAOSTest no longer run 2.0 testing since 2.0 is EOL?

**Comment by brianjmurrell** [comment]
> Side note: @brianjmurrell Should packageBuildingPipelineDAOSTest no longer run 2.0 testing since 2.0 is EOL?

Probably.  Another thing on the pile.  There's a lot of stuff that should be discontinued with 2.0 being EOL.

**Comment by daltonbohning** [comment]
> > A single 2.2 test failed: https://build.hpdd.intel.com/blue/organizations/jenkins/daos-stack%2Fdaos/detail/ci-daos-stack-mpifileutils-pkg-PR-30-release-2.2/4/tests For a known reason: https://daosio.atlassian.net/browse/DAOS-10258
> 
> Interesting that it's run when the ticket for it says it's in the skip-list. Before I start asking elsewhere, do you have any idea why @daltonbohning?

It's because the issue was "resolved", the skiplist updated, and then the issue re-opened without the skiplist being updated again. I pinged Phil on the ticket to see if the skiplist should be updated again

**Review by brianjmurrell — APPROVED** [review]

**Comment by brianjmurrell** [comment]
> daltonbohning requested a review from daos-stack/build-and-release-watchers 2 hours ago

@daltonbohning Did you mean to request review from @daos-stack/release-engineering it indicate a request for landing?  Just want to make sure it's your intention to request landing before going ahead and doing it.

**Comment by daltonbohning** [comment]
> @daltonbohning Did you mean to request review from @daos-stack/release-engineering it indicate a request for landing? Just want to make sure it's your intention to request landing before going ahead and doing it.

@brianjmurrell Yes. Thanks and sorry :)

**Comment by brianjmurrell** [comment]
> @brianjmurrell Yes. Thanks and sorry :)

No worries.  Just wanted to confirm.

**Pull request merged**
